src/Protocol/ForgeHandshake.cpp

```diff
@@ -0,0 +1,365 @@
+
+// ForgeHandshake.cpp
+
+// Implements Forge protocol handshaking
+
+#include "Globals.h"
+#include "ForgeHandshake.h"
+#include "json/json.h"
+#include "../Server.h"
+#include "../Bindings/PluginManager.h"
+#include "../ClientHandle.h"
+#include "../Root.h"
+
+
+/** Discriminator byte values prefixing the FML|HS packets to determine their type. */
+enum
+{
+	Discriminator_ServerHello = 0,
+	Discriminator_ClientHello = 1,
+	Discriminator_ModList = 2,
+	Discriminator_RegistryData = 3,
+	Discriminator_HandshakeReset = -2,
+	Discriminator_HandshakeAck = -1,
+};
```

Comment:
enum class still fails on gcc :( although clang is fine, this causes the Travis-CI build to fail: https://travis-ci.org/cuberite/cuberite/jobs/258935845

/home/travis/build/cuberite/cuberite/src/Protocol/ForgeHandshake.cpp: In member function ‘void cForgeHandshake::DataReceived(cClientHandle*, const char*, size_t)’:
/home/travis/build/cuberite/cuberite/src/Protocol/ForgeHandshake.cpp:343:26: error: ‘Discriminator’ is not a class, namespace, or enumeration
   case static_cast<Int8>(Discriminator::ClientHello):
                          ^
/home/travis/build/cuberite/cuberite/src/Protocol/ForgeHandshake.cpp:347:26: error: ‘Discriminator’ is not a class, namespace, or enumeration
   case static_cast<Int8>(Discriminator::ModList):
                          ^
/home/travis/build/cuberite/cuberite/src/Protocol/ForgeHandshake.cpp:351:26: error: ‘Discriminator’ is not a class, namespace, or enumeration
   case static_cast<Int8>(Discriminator::HandshakeAck):
                          ^
I'll have to revert it unless we can find a compatible solution...